lighthouse-core/gather/gatherers/seo/tap-targets-test.js

```diff
@@ -0,0 +1,152 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+const assert = require('assert');
+const {getBCRs, simplifyBCRs} = require('./tap-targets');
+
+describe('SEO: Tap targets gatherer', () => {
+  // describe('getBCRs', () => {
+  //   it('For a small link that contains a large image it includes the image bcr', () => {
+  //     // i guess if one bcr contains another, or they are the same, then merge them
+  //     const linkNode = {};
+  //     getBCRs(linkNode);
+  //   });
+  // });
+
+  describe('simplifyBCRs', () => {
+    it('Merges BCRs if a smaller BCR is inside a larger one', () => {
+      const res = simplifyBCRs([
+        {
+          left: 10,
+          right: 110,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+        {
+          left: 10,
+          right: 60,
+          width: 50,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+      ]);
+      assert.deepEqual(res, [
+        {
+          left: 10,
+          right: 110,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+      ]);
+    });
+    it('Merges two horizontally adjacent client rects', () => {
+      const res = simplifyBCRs([
+        {
+          left: 10,
+          right: 110,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+        {
+          left: 110,
+          right: 210,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+      ]);
+      assert.deepEqual(res, [
+        {
+          left: 10,
+          right: 210,
+          width: 200,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+      ]);
+    });
+
+    it('Merges three horizontally adjacent client rects', () => {
+      const res = simplifyBCRs([
+        {
+          left: 10,
+          right: 110,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+        {
+          left: 110,
+          right: 210,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+        {
+          left: 210,
+          right: 310,
+          width: 100,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+      ]);
+      assert.deepEqual(res, [
+        {
+          left: 10,
+          right: 310,
+          width: 300,
+          height: 10,
+          top: 10,
+          bottom: 20,
+        },
+      ]);
+    });
+
+
+    it('Merges two vertically adjacent client rects', () => {
+      const res = simplifyBCRs([
```

Comment:
before you write too many more tests, I might make a helper function that just lets you pass in `{x, y, width, height}` to make these a little easier to understand.

I dunno about you but I've always found width, height much easier to quickly parse than top, left, right, bottom :)

Reply:
Oh, good idea!